app/src/modules/settings/routes/roles/app-permissions.ts

```diff
@@ -121,6 +121,7 @@ export const appRecommendedPermissions: Partial<Permission>[] = [
 			'avatar',
 			'language',
 			'theme',
+			'tfa_secret',
```

Comment:
This does resolve #9883 for new setups, but existing setups will already have the recommended permissions _without_ this field, thus getting error 403 when entering their OTP in the enforce tfa flow.

Perhaps we'll need to add a notice/warn about this 🤔

Reply:
I think the API endpoint to enable/disable TFA shouldn't rely on regular user permissions. It should require the user's ID, current user's password, and then allow updating tfa_secret without relying on the role permissions.

Reply:
Hmm that's also true, i'm just really worried about the annoyance / confusion that is requiring the explicit update permission to this one field. We'd also don't want to expose updating that particular field for _other_ users, so if you're not paying close attention to the rules configured for the role, you end up allowing any user to set the tfa secret for other users 🤔 I think TFA configuration might be an exception to our regular permissions handling, as it's such a security sensitive topic

Reply:
Awesome, will look into patching this up later. 👍🏻

Reply:
@rijkvanzanten No worries 😂 I completely understand these are two different problems with very different requirements.

Reply:
Just noticed that it was already mentioned in the docs https://github.com/directus/directus/pull/7805/commits/5a129b6815cc4ff1fc3319cec874303f6f37dd8e 🙈  
Note to revert the docs change after the patch.

Reply:
Done.